### ISSUE TITLE
Slideshow not working

### ISSUE THREAD

**Issue by edgaradr223** [AlexIIL/BetterLoadingScreen#73]
Hello guys, im trying to create a slideshow following the instructions on the wiki for minecraft 1.12.2 using CLS 1.5.1,  the problem is that when i launch minecraft it shows this instead:

![image](https://user-images.githubusercontent.com/50599044/136316654-7a568c71-5a1c-440e-8167-eca823df18fc.png)

im sure everything is set as it is supposed to.... is this a known issue? or am i missing something?

**Comment by AlexIIL** [comment]
That screen shows up when something goes very wrong while loading CLS - hopefully your log file contains more details?

(Although you may want to try updating CLS to [1.5.5](https://www.curseforge.com/minecraft/mc-mods/better-loading-screen/files/3337214) first, since that's a bit newer than 1.5.1)

**Comment by edgaradr223** [comment]
I managed to fix it Alex, what i did was setting sample/slideshow instead of config/slideshow in cls.cfg, then i opened your mod .jar and changed the slideshow conditions from that file instead of the slideshow.json in config and now it runs perfectly, excellent mod, very good job and congratulations, only bad thing is for some reason Im consuming 6gb ram memory instead of 4.5 i was using before, but i suppose this is the price to pay for such masterpiece.

Thank you :)

**Comment by AlexIIL** [comment]
> then i opened your mod .jar and changed the slideshow conditions from that file instead of the slideshow.json

er, what? You're not meant to need to do that. At least it worked though :p

The extra 1.5 GB of memory is a bit more worrying though - does that RAM usage go down after CLS has finished? If not then I'll need to actually clear out memory after CLS has finished using it...

**Comment by edgaradr223** [comment]
The RAM usage has gone down from 6.2gb to 5.3gb after I removed the half of the .png files i loaded first (was using 54, now using 25) but as i can see that RAM gap remains even when CLS has finished using it, i mean ... not big deal for me, I play minecraft with 16gb RAM allocated but it may be a problem to users with low pc specs (8gb or less and outdated or old processors)

> er, what? You're not meant to need to do that. At least it worked though :p

It seems like both 1.5.1 and 1.5.5 versions are ignoring reading files when set to config/slideshow and your file is located at /customloadingscreen/slideshow.json, maybe a mod or forge compatibility issue? I dont know im just a newbie XD, thats why i tried to change the parameters from the jar file and adding sample/slideshow worked, hope it serves as reference for anyone who comes here with the same issue.

**Comment by edgaradr223** [comment]
hey mate, any idea why is this happening? the game crashes on opening every 4 or 5 times i launch minecraft and its always when cls reaches 100% load

Stacktrace:
	at net.minecraftforge.fml.client.SplashProgress.checkThreadState(SplashProgress.java:601)
	at net.minecraftforge.fml.client.SplashProgress.pause(SplashProgress.java:614)
	at net.minecraft.client.Minecraft.func_71384_a(Minecraft.java:522)

-- Initialization --
Details:
Stacktrace:
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:378)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

**Comment by AlexIIL** [comment]
Ok, the extra RAM usage post-launch should be fixed in 1.5.6.

In addition I've added an "example.json" config file that you can use to test loading from - if you change the "screen_config" to "config/example" then it should load an example loading screen - and if so it should be easier to find where to place your own configs so that CLS will load them.

**Comment by edgaradr223** [comment]
Ok, i think now i know what the issue may be with slideshow, apparently all the other configs work quite right as you can see with these 2 examples using sample/worldload and sample/scrolling:

![scrolling](https://user-images.githubusercontent.com/50599044/136678135-18d4d495-2ea1-4a07-9f06-3caf3c3f9df7.jpg)
![worldload](https://user-images.githubusercontent.com/50599044/136678136-1074a703-c256-4d36-977b-a7a82d4768ac.jpg)

Now, what differences them from slideshow is that slideshow needs to load .png from a different directory, tried to load my .png files opening your mod and adding them to cls.jar/assets/customloadingscreen but it keeps throwing the messy window, so what i think is that maybe something is making the config file to load slideshow.json but not the .png files

**Comment by edgaradr223** [comment]
oh it also works using config/example by the way

**Comment by AlexIIL** [comment]
Huh, that's odd. Can you upload your log file from when CLS doesn't find your screenshot files? That should contain some details as to what's going wrong.

**Comment by edgaradr223** [comment]
[2021-10-09-7.log](https://github.com/AlexIIL/CustomLoadingScreen/files/7316696/2021-10-09-7.log)

this is my last log, i have no idea how to read that

**Comment by AlexIIL** [comment]
Thanks!

````
 java.lang.NullPointerException
	at sun.reflect.UnsafeFieldAccessorImpl.ensureObj(UnsafeFieldAccessorImpl.java:57)
 	at sun.reflect.UnsafeQualifiedObjectFieldAccessorImpl.get(UnsafeQualifiedObjectFieldAccessorImpl.java:38)
	at java.lang.reflect.Field.get(Field.java:393)
	at alexiil.mc.mod.load.render.TextureLoader.openResourceStream(TextureLoader.java:162)
	at alexiil.mc.mod.load.json.subtypes.JsonRenderSlideshow.actuallyBake(JsonRenderSlideshow.java:56)
	at alexiil.mc.mod.load.json.subtypes.JsonRenderSlideshow.actuallyBake(JsonRenderSlideshow.java:27)
````

This is the useful part of the log - and indicates I'm reading something incorrectly, so I'll have a look at that.

**Comment by AlexIIL** [comment]
Ok, I think I've fixed that in 1.5.7-pre.1 - since I don't want to spam curseforge with testing releases I've pushed it to my site: https://alexiil.uk/mods/CustomLoadingScreen/CustomLoadingScreen-1.12.2-1.5.7-pre.1.jar)

**Comment by edgaradr223** [comment]
its working properly now, tried from config/slideshow and from sample/slideshow both works, opened/closed minecraft nearly 11 times and no crash and also im back to my 4.5gb ram once the game begins, amazing job Alex! cant wait to see the rest of your mods, you became my favourite developer and i really appreciate your patience and your will to help.

![image](https://user-images.githubusercontent.com/50599044/136679633-c8d6dfc9-65fc-4bbf-9e51-bcc56adce9dd.png)

**Comment by AlexIIL** [comment]
Thanks! I'll release 1.5.7 soon, probably after adding a few more bugfixes :)

**Comment by Exzept1on** [comment]
> Ok, I think I've fixed that in 1.5.7-pre.1 - since I don't want to spam curseforge with testing releases I've pushed it to my site: https://alexiil.uk/mods/CustomLoadingScreen/CustomLoadingScreen-1.12.2-1.5.7-pre.1.jar)

Hello. Can you publish version 1.5.7, since there is still a crash on 1.5.6, which is described in this error, but it is not in 1.5.7-pre1, I need this version on curseforge. Thanks.

**Comment by gnomed-by-steve** [comment]
> > since I don't want to spam curseforge with testing releases

> Hello. Can you publish version 1.5.7

haa**aUUUUUUUGH**

**Comment by AlexIIL** [comment]
> haaaUUUUUUUGH

10/10, best notification.

I've published [1.5.7](https://www.curseforge.com/minecraft/mc-mods/better-loading-screen/files/3769828) to curseforge - sorry for the delay!
